### PR TITLE
Java front-end: remove unused parameters

### PR DESCRIPTION
**Pull request by tautschnig**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 8212285f910a02af088f1cc8587f67d7de06f5bc).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/77351099
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

File: jbmc/src/java_bytecode/select_pointer_type.h [hunk 1]
```diff
@@ -39,8 +39,7 @@ class select_pointer_typet
   virtual pointer_typet convert_pointer_type(
     const pointer_typet &pointer_type,
     const generic_parameter_specialization_mapt
-      &generic_parameter_specialization_map,
-    const namespacet &ns) const;
```
Comment (peterschrammel) [inline jbmc/src/java_bytecode/select_pointer_type.h]: Parameter is required in derived classes that override this method.

**Review by peterschrammel — CHANGES_REQUESTED** [review]
and needs a rebase

File: jbmc/src/java_bytecode/select_pointer_type.h (hunk 1 above)
Comment (tautschnig) [reply jbmc/src/java_bytecode/select_pointer_type.h]: Parameter restored and documented.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 87e9f74770cc4fc6fa04dede80f03ab0adfa3675).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78386656

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 75855bf210e7b226675b6f42740e879bb888b837).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80943269

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**